### PR TITLE
Auth & User Data Implementation

### LINKED ISSUE #7: [5pts] App Backend: User login
**Fill in the blanks**:
As a student, I would like to login to the app, so that I can sync/save my schedule and add other users as friends.

**Definition of Done - Check as you go!**
- [ ] Tasks have been implemented
- [ ] Documented
- [ ] Passing Tests for each task
- [ ] Pull Request made
- [ ] Code Reviewed

### LINKED ISSUE #121: [5pts] (Front/Back)end: Connect Register and Login Page
**Fill in the blanks**:
As a user, I would like to login/register, so that my account is stored and accessible.

**Describe the solution you'd like:**
- [x] Connect register to backend
- [ ] Connect login to backend
- [ ] Use database to start storing login/register pages

**Definition of Done - Check as you go!**
- [ ] Tasks have been implemented
- [x] Documented
- [ ] Passing Tests for each task
- [ ] Pull Request made
- [ ] Code Reviewed

### LINKED ISSUE #133: [13pts] Connect friends frontend to backend
**Fill in the blanks**:
This task will be completed when the frontend and backend for the friends feature work seamlessly.

**How frequent is this task?**:
- ( ) Weekly
- (X) Once

### PR DESCRIPTION
**Pull request by RafaelPiloto10**
This PR implements complete user auth for the app, user data initialization, and the where-am-i backend for grabbing friend data. Does not implement updating user where-am-i data as blocked by #153

### CODE REVIEW

File: quack_app/lib/core/food/menu_data.dart [hunk 1]
```diff
@@ -10,11 +10,148 @@ class MenuData {
...
-  
+  Map<String, Map<String, Object>> testData = {
+    "stations": {
```
Comment (RafaelPiloto10) [inline quack_app/lib/core/food/menu_data.dart]: VSCode auto-formatted -- sorry

File: quack_app/lib/core/auth/auth.dart [hunk 2]
```diff
@@ -61,31 +63,49 @@ class Auth {
...
-          print("[core/auth] Successfully logged in");
-          return Future.value(true);
-        }
-        print("[core/auth] Failed to login ${response.body}");
+  Future<List> authenticate(String email, password) async {
+    if (istest) {
+      return Future.value([
+        "",
+        {"access_token": ""},
+        jsonEncode({
```
Comment (brendacano) [inline quack_app/lib/core/auth/auth.dart:71]: this is pretty cool that it stores all this information inside the encoding

**Review by brendacano — APPROVED** [review]
I didn't find anything that should be changed or could be improved on. It looks good, nice work dude.

**Review by DDVD233 — APPROVED** [review]
Great work! I was again repeating some of your works (on the front end side)...

**Pull request merged**